lib/services/database/photo_data_entry.dart

```diff
@@ -0,0 +1,96 @@
+import 'dart:ffi';
+
+import 'package:mobileprism/services/database/database_service.dart';
+
+class PhotoDataEntry {
+  late final String uid;
+  late final bool? panorama;
+  late final int? width;
+  late final int? height;
+  late final String? imageHash;
+  late final String? imageQuality;
+  late final double? lat;
+  late final double? long;
+  late final Int64? timestamp;
+
+  PhotoDataEntry({
+    required this.uid,
+    this.panorama,
+    this.width,
+    this.height,
+    this.imageHash,
+    this.imageQuality,
+    this.lat,
+    this.long,
+    this.timestamp,
+  });
+
+  PhotoDataEntry.fromMap(Map<String, Object?> data) {
```

Comment:
Have a look at [this](https://docs.flutter.dev/development/data-and-backend/json#serializing-json-inside-model-classes).

I think there is no need for exceptions here. If `uid` is missing, you might get an exception anyway because it is requiered.